_summaries/shi2018pointrcnn_2.md

```diff
@@ -1,19 +1,124 @@
+    </p>
+
+    $$N_{\text{pos}}$$ is the number of foreground points.
+
+    For inference on bin-based parameters $$(x, z, \theta)$$, the model first chose the bin center with the highest predicted confidence, and then added the predicted residuals to obtain the refined parameters. For inference on directly regressed parameters $$(y, h, w, l)$$, the model just add the predicted residuals.
+
+    To remove redundant proposals, the authors used non-maximum suppression (NMS) with some IoU threshold. The authors also kept the top 300 proposals for training the 2nd stage, and kept the top 100 proposals for inference on the 2nd stage.
+
+**Stage 2: Point Cloud RoI Pooling and Canonical 3D Bounding Box Refinement**
+
+The bounding boxes collected from stage 1 will be slightly enlarged. All points within the bounding box will be used for the bounding box refinement later. In addition, the points within the bounding box will go through a canonical transformation so that 1) the center of the bounding box is at the origin, 2) the $$X'$$ axis is pointing towards the head direction of the proposal with the $$Z'$$ axis perpendicular to it, and 3) the $$Y'$$ axis remains the same as that of the LIDAR coordinate system:
+
+<p align="center">
+    <img src="shi2018pointrcnn/canonical_transform.png" width="30%" height="30%">
+</p>
+
+For each point, the 3D point coordinates, the laser reflection intensity, predicted segmentation mask, and a distance of how far the point is from the origin will be concatenated together to form local features. The distance feature is used to compensate for the lost depth information for far-away objects, which generally have much fewer points in LIDAR sensors than nearby objects. The local features will then be transformed through an MLP to get the same dimension as the semantic point features (from the point network). These are concatenated and fed to another network to obtain a confidence classification and box refinement.
+
+A ground-truth box is matched with a proposal if their IoU is above 0.55. The ground-truth box also goes through the canonical transformation. To do refinement, the proposal's formulation is the same as in stage 1, but with smaller bins for proposal center and orientation. The refinement loss is defined as:
+
+<p align="center">
+    <img src="shi2018pointrcnn/refinement_loss.png" width="20%" height="20%">
+</p>
+
+Where $$\mathcal{B}$$ is the set of 3D proposals from stage-1 and $$\mathcal{B}_{pos}$$ are the positive (>0.55 IoU) proposals for regression. $$\text{prob}_{i}$$ denotes the estimated confidence of the bounding box. Eventually, another NMS is applied to remove overlapping bounding boxes with an IoU threshold of 0.01.
+
 * What interesting variants are explored?
 
+**GT-AUG**: To simulate objects with various environments, the authors copied several ground-truth boxes and their inside points from some scenes to same locations of other scenes. This slightly improved mAP on the validation set.
```

Comment:
Consider briefly mentioning the other data augmentations the authors used (random flip, rotation, etc.) as well as other implementation details (ex. ~16k points were subsampled from each point cloud with repeating points if the input is smaller).